src/index.rs

```diff
@@ -182,17 +180,15 @@ impl Index {
                     )
                 }
                 let blockhashes: Vec<BlockHash> = chunk.iter().map(|h| h.hash()).collect();
-                let mut heights_map: HashMap<BlockHash, usize> =
-                    chunk.iter().map(|h| (h.hash(), h.height())).collect();
+                let mut heights = chunk.iter().map(|h| h.height());
 
                 let mut batch = WriteBatch::default();
-
-                daemon.for_blocks(blockhashes, |blockhash, block| {
-                    let height = heights_map.remove(&blockhash).expect("unexpected block");
+                daemon.for_blocks(blockhashes, |_blockhash, block| {
+                    let height = heights.next().expect("unexpected block");
```

Comment:
Not sure if panicking because external input is unexpected is the best course of action but I find it understandable if you believe it is.